lightning/src/ln/channelmanager.rs

```diff
@@ -3097,9 +3123,8 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 
 			for (short_chan_id, mut pending_forwards) in channel_state.forward_htlcs.drain() {
 				if short_chan_id != 0 {
-					let forward_chan_id = match channel_state.short_to_chan_info.get(&short_chan_id) {
-						Some((_cp_id, chan_id)) => chan_id.clone(),
-						None => {
+					macro_rules! fail_pending_forwards {
+						() => {
```

Comment:
Think you can get rid of the macro with something like:
```suggestion
					let forward_chan_id_opt = match self.short_to_chan_info.read().unwrap().get(&short_chan_id)
						.map(|(_cp_id, chan_id)| chan_id.clone());
					if forward_chan_id_opt.is_none() || !channel_state.by_id.contains_key(forward_chan_id_opt.unwrap()) {

```
but I haven't actually completed this patch myself

Reply:
Oh, thanks for the suggestion :)! 

That change would indeed work when only taking this PR into account. Sadly though, that change would not help when taking #1507 into consideration as well, which I'd like this PR to prepare for. 1507 is quite huge, so I'd like to do as many changes as possible within this PR, to make 1507 as reviewable as possible.

In 1507, we will lock the new `channel lock` using in this scope by using the info retrieved from `short_to_chan_info` (The `counterparty_node_id` and the `channel_id`). Therefore we wouldn't be able to check that `short_to_chan_info` both contained a value for the `scid` and check that the new `channel lock` contains the channel for the unwrapped `short_to_chan_info` info in the same `if` clause, unless we'd take the new `channel lock` temporarily just for the contains check. Though by taking the `channel lock` temporarily, we would still need the `Vaccant` check below for the `channel lock `, hence requiring the second call for the code in `fail_pending_forwards` :/

Reply:
Ah okay, worth a shot!